### DISCUSSION TITLE
DDM for Conflict Tasks e.g. Flanker in HSSM?

### DISCUSSION THREAD

**Discussion by acburns** [lnccbrown/HSSM#529] [General]
Hello, 

I am wondering if there are plans or if anyone has implemented any models for dealing with conflict tasks like the Flanker task, e.g. the diffusion model for conflict tasks? 

Thanks!

Angus

**Comment by frankmj** [comment]
In fact we have a couple of these in the works! Just doing some testing on
identifiability issues etc and hopefully one or two of these models will be
available for others to try within a month.

Michael J Frank, PhD | Edgar L. Marston Professor
Director, Carney Center for Computational Brain Science
<https://www.brown.edu/carney/ccbs>
Laboratory of Neural Computation and Cognition <https://www.lnccbrown.com/>

Brown University
website <http://ski.clps.brown.edu>

**Comment by acburns** [comment]
Thanks for the update! Look forward to trying them out.

**Comment by acburns** [comment]
Hello!

Just wondering if there is still a plan to add conflict (e.g. Flanker) models?

Thanks!

Angus

**Comment by AlexanderFengler** [comment]
Hi @acburns , 

testing on identifiability issues turned out to be quite a lot more work than we had anticipated, and we needed to transfer the labor between people because of ending contracts, but your timing is good. 

We recently closed the loop on that and have a version of [the](https://pmc.ncbi.nlm.nih.gov/articles/PMC3195995/) 'shrinking spotlight' model that is identifiable, trained as a LAN. 

We are putting some final touches on that and will be in a position to share a tutorial notebook and upload the model to Huggingface to have it available in base HSSM, within a week. 

I will set this as a strict internal deadline and ping you if I foresee any further delays.

**Comment by AlexanderFengler** [comment]
Hey @acburns, 

working on this actively, taking slightly longer than anticipated.

Best,
Alex

**Comment by AsgeirsdottirUA** [comment]
Hi all,

Just wanted to follow up on this thread - I'm interested in fitting a DDM conflict model, so I wanted to check on the current status of this.

1. What are the current options for fitting conflict models with HSSM? I've recently come across ssms/config/_modelconfig/conflict.py and ssms/config/_modelconfig/gamma_drift.py, but these aren't in hssm.HSSM.supported_models and I can't seem to find any trained LAN/ONNX for the conflict models in conflict.py. What would be the recommended approach for fitting these models (and is there any work in progress on trained LANs for these conflict models)?
2. Are there any tutorials planned or available for working with conflict models in HSSM? A worked example would help a lot.

I also came across what looks like a trained LAN for gamma-drift here: https://github.com/AndrewZhang599/conflict-tasks/tree/main/trained%20LAN%20data/onnx%20models -- not sure if it's considered validated.

Best,
Andrea

**Comment by AlexanderFengler** [comment]
@AsgeirsdottirUA,

We are implementing a `spotlight model` ( @AndrewZhang599 ) and a few basic conflict DDMs (e.g. gamma distorted drift), which will be backfilled and made available officially (through the HuggingFace repository).

Development took a lot of detours since this discussion was openend, however the road is now 100% clear for this and I will add the basic conflict DDMs myself.

We are also now working on model-wise tutorials which will be made available through the documentation of the [ssm-simulators](https://github.com/lnccbrown/ssm-simulators) package.

If there is a specific model you would like to use, consider sending a link to the corresponding paper.

Best,
Alex
